controller/customdomain_utils.go

```diff
@@ -104,6 +114,83 @@ func FindCustomDomainCondition(conditions []customdomainv1alpha1.CustomDomainCon
 	return nil
 }
 
+// Take an ingress controller managed by the custom domains operator and release it back to the
+// cluster ingress operator. Also schedule it onto customer worker nodes from the Red Hat managed infra
+// nodes.
+func (r *CustomDomainReconciler) returnIngressToClusterIngressOperator(reqLogger logr.Logger, instance *customdomainv1alpha1.CustomDomain) (ctrl.Result, error) {
+	reqLogger.Info(fmt.Sprintf("Removing operator management labels from %s's underlying ingress controller", instance.Name))
+
+	ingressName := instance.Name
+	customIngress := &operatorv1.IngressController{}
+
+	reqLogger.Info(fmt.Sprintf("Fetching ingress controller: %s/%s", ingressOperatorNamespace, ingressName))
+	err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: ingressOperatorNamespace,
+		Name:      ingressName,
+	}, customIngress)
+
+	if err != nil {
+		if kerr.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Error(err, fmt.Sprintf("Ingresscontroller %s in %s namespace not found", ingressName, ingressOperatorNamespace))
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	delete(customIngress.Labels, managedLabelName)
+	customIngress.Spec.NodePlacement = &operatorv1.NodePlacement{
+		NodeSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{"node-role.kubernetes.io/worker": ""},
+		},
+		Tolerations: []corev1.Toleration{},
+	}
+
+	reqLogger.Info(fmt.Sprintf("Updating ingress %s with new node placement on worker node, removing tolerations for infra nodes", instance.Name))
+	err = r.Client.Update(context.TODO(), customIngress)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Error updating ingresscontroller %s in %s namespace", ingressName, ingressOperatorNamespace))
+		return reconcile.Result{}, err
+	}
```

Comment:
I've got a feeling it will recreate the pods immediately on worker nodes  once 'update' fires - but we need to verify the behaviour to make sure that it's doing it in a graceful way (without network disruption).

Reply:
We could delete the router's deployment prior to updating in order to start off fresh. This will, obviously, have the side effect of creating some downtime when handing off to the ingress operator, though.

Reply:
If this turns out not to gracefully reschedule, we can create a temporary PDB within the reconcile loop for the lifetime of the reschedule. Let's hope we don't have to go down that road.